source/common/router/config_impl.cc

```diff
@@ -429,28 +442,13 @@ RouteEntryImplBase::loadRuntimeData(const envoy::api::v2::route::RouteMatch& rou
   RuntimeData runtime_data;
 
   if (route_match.runtime_specifier_case() == envoy::api::v2::route::RouteMatch::kRuntimeFraction) {
-    envoy::type::FractionalPercent fractional_percent;
-    const std::string& fraction_yaml =
-        loader_.snapshot().get(route_match.runtime_fraction().runtime_key());
-
-    try {
-      MessageUtil::loadFromYamlAndValidate(fraction_yaml, fractional_percent);
-    } catch (const EnvoyException& ex) {
-      ENVOY_LOG(error, "failed to parse string value for runtime key {}: {}",
-                route_match.runtime_fraction().runtime_key(), ex.what());
-      fractional_percent = route_match.runtime_fraction().default_value();
-    }
-
-    runtime_data.numerator_val_ = fractional_percent.numerator();
-    runtime_data.denominator_val_ =
-        ProtobufPercentHelper::fractionalPercentDenominatorToInt(fractional_percent.denominator());
+    runtime_data.fractional_runtime_default_ = route_match.runtime_fraction().default_value();
+    runtime_data.fractional_runtime_key_ = route_match.runtime_fraction().runtime_key();
+    runtime_data.legacy_runtime_data_ = false;
   } else if (route_match.runtime_specifier_case() == envoy::api::v2::route::RouteMatch::kRuntime) {
-    // For backwards compatibility, the deprecated 'runtime' field must be converted to a
-    // RuntimeData format with a variable denominator type. The 'runtime' field assumes a percentage
-    // (0-100), so the hard-coded denominator value reflects this.
-    runtime_data.denominator_val_ = 100;
-    runtime_data.numerator_val_ = loader_.snapshot().getInteger(
-        route_match.runtime().runtime_key(), route_match.runtime().default_value());
+    runtime_data.runtime_default_ = route_match.runtime().default_value();
+    runtime_data.runtime_key_ = route_match.runtime().runtime_key();
+    runtime_data.legacy_runtime_data_ = true;
```

Comment:
Is it possible to deal with this using an optional vs. having an additional boolean? I realize this code is about to get deleted in a follow up for the deprecation, so up to you.

Reply:
I plan to rip all of this out almost immediately after merging this.

Reply:
Even more fun would be https://en.cppreference.com/w/cpp/utility/variant. You can use `absl::variant` even before C++17 :)